examples/window-app/common/src/WindowApp.cpp

```diff
@@ -296,33 +293,54 @@ void WindowApp::DispatchEvent(const WindowApp::Event & event)
             GetCover().LiftDown();
         }
         break;
-
-    case EventId::LiftUp:
-    case EventId::LiftDown:
-        cover = GetCover(event.mEndpoint);
-        if (cover)
-        {
-            cover->GotoLift(event.mId);
-        }
+    case EventId::AttributeChange:
+        DispatchEventAttributeChange(event.mEndpoint, event.mAttributeId);
         break;
-
-    case EventId::TiltUp:
-    case EventId::TiltDown:
-        cover = GetCover(event.mEndpoint);
-        if (cover)
-        {
-            cover->GotoTilt(event.mId);
-        }
+    default:
         break;
+    }
+}
 
-    case EventId::StopMotion:
-        cover = GetCover(event.mEndpoint);
-        if (cover)
-        {
-            cover->StopMotion();
-        }
-        break;
+void WindowApp::DispatchEventAttributeChange(chip::EndpointId endpoint, chip::AttributeId attribute)
+{
+    Cover * cover = GetCover(endpoint);
+
+    if (nullptr == cover)
+    {
+        emberAfWindowCoveringClusterPrint("Ep[%u] not supported AttributeId=%u\n", endpoint, (unsigned int) attribute);
+        return;
+    }
 
+    switch (attribute)
+    {
+    /* For a device supporting Position Awareness : Changing the Target triggers motions on the real or simulated device */
+    case Attributes::TargetPositionLiftPercent100ths::Id:
+        cover->LiftGoToTarget();
```

Comment:
As a followup: this seems like it should be shared cluster logic, not per-app logic...  That is, the cluster should notice the attribute change and trigger an app callback of some sort, instead of every app having to know how attributes map into behavior.

Reply:
indeed we need to have single implementation as a follow up
but due to amount of diff and time constraint my goal is to implement all the missing test in the TH